### DISCUSSION TITLE
Is there a way to detect tcp broken connection faster?

### DISCUSSION THREAD

**Discussion by qishiyexu** [libuv/libuv#3370] [Q&A]
When a tcp connection is broken, I can get notify from read_cb, but it is slow.
Here is my code:

```
uv_tcp_keepalive(&conn->handle.tcp, 1, 3);
uv_read_start(&conn->handle.stream,
                                 alloc_cb,
                                 read_cb);

read_cb(uv_stream_t* handle, ssize_t nread, const uv_buf_t* buf)
{
// get notify
}
```
When I disable my network, read_cb will be called and nread is -60(connection timed out), however it's too slow, I had to  wait 30+ seconds then reab_cb was called.  
Is there a way to make it called faster?

**Comment by bnoordhuis** [comment]
The short answer is "no" - TCP is designed to be robust in the face of temporary network outages. There are [kernel knobs you can tweak](https://www.man7.org/linux/man-pages/man7/tcp.7.html)  but they're not portable and outside libuv's remit.

**Reply by qishiyexu** [reply]
But I using glib network lib I can get a much more faster notify, does libuv deliberately slow it down?

**Comment by bnoordhuis** [comment]
It doesn't.
